node_modules/oae-authz/lib/api.js

```diff
@@ -1120,6 +1120,11 @@ var _canExplicitlyInteract = function(ctx, resourceTenantAlias, targetResources,
     // explicit access for users. The role priority list is not really important as we aren't trying
     // to derive an exact role
     if (AuthzUtil.isUserId(targetResource.id)) {
+        // You can always interact with guest users
```

Comment:
I'm not entirely sure either. If they would be allowed to have guests, I would think that those guests would need to be on the same tenant, otherwise we'd start to violate the can interact rules. I think it's best to just disallow guest sharing for private tenants.